app/api/slack/channels/route.ts

```diff
@@ -0,0 +1,68 @@
+import { NextResponse } from "next/server";
+import { createErrorResponse } from "@/utils/error-response";
+
+export async function GET() {
+  const requestId = `slack-channels-${Date.now()}`;
+  
+  try {
+    const slackBotToken = process.env.SLACK_BOT_TOKEN;
+    if (!slackBotToken) {
+      console.error(`[${requestId}] SLACK_BOT_TOKEN environment variable is not set`);
+      return NextResponse.json(
+        createErrorResponse("Missing Slack bot token", {
+          code: "MISSING_TOKEN",
+          details: "SLACK_BOT_TOKEN environment variable is required",
+          requestId,
+          troubleshooting: [
+            "Set SLACK_BOT_TOKEN environment variable",
+            "Check your environment configuration",
+            "Ensure the token has proper scopes"
+          ]
+        }),
+        { status: 500 }
+      );
+    }
+
+    const response = await fetch("https://slack.com/api/conversations.list", {
+      method: "GET",
+      headers: {
+        "Authorization": `Bearer ${slackBotToken}`,
+        "Content-Type": "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Slack API error: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+    
+    if (!data.ok) {
+      throw new Error(`Slack API error: ${data.error}`);
+    }
+
+    const channels = data.channels.map((channel: { id: string; name: string; is_private?: boolean }) => ({
+      id: channel.id,
+      name: channel.name,
+      isPrivate: channel.is_private || false,
+      unreadCount: 0
+    }));
+
+    return NextResponse.json({ channels });
```

Comment:
Am failing with this error

![image](https://github.com/user-attachments/assets/12baa04f-9317-47c8-8665-f9b027253f4f)

Return more info in the error case so that we could see what's wrong